streaming/java/streaming-runtime/src/main/java/io/ray/streaming/runtime/master/JobMaster.java

```diff
@@ -0,0 +1,130 @@
+package io.ray.streaming.runtime.master;
+
+import com.google.common.base.Preconditions;
+import io.ray.api.RayActor;
+import io.ray.streaming.jobgraph.JobGraph;
+import io.ray.streaming.runtime.config.StreamingConfig;
+import io.ray.streaming.runtime.config.StreamingMasterConfig;
+import io.ray.streaming.runtime.core.graph.executiongraph.ExecutionGraph;
+import io.ray.streaming.runtime.master.graphmanager.GraphManager;
+import io.ray.streaming.runtime.master.graphmanager.GraphManagerImpl;
+import io.ray.streaming.runtime.master.resourcemanager.ResourceManager;
+import io.ray.streaming.runtime.master.resourcemanager.ResourceManagerImpl;
+import io.ray.streaming.runtime.master.scheduler.JobSchedulerImpl;
+import io.ray.streaming.runtime.worker.JobWorker;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JobMaster is the core controller in streaming job as a ray actor. It is responsible for all the
+ * controls facing the {@link JobWorker}.
+ */
+public class JobMaster {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JobMaster.class);
+
+  private JobRuntimeContext runtimeContext;
+  private ResourceManager resourceManager;
+  private JobSchedulerImpl scheduler;
+  private GraphManager graphManager;
+  private StreamingMasterConfig conf;
+
+  private RayActor jobMasterActor;
+
+  public JobMaster(Map<String, String> confMap) {
+    LOG.info("Job master init with conf: {}.", confMap);
+
+    StreamingConfig streamingConfig = new StreamingConfig(confMap);
+    this.conf = streamingConfig.masterConfig;
+
+    // init runtime context
+    runtimeContext = new JobRuntimeContext(streamingConfig);
+
+    LOG.info("Job master init success.");
+  }
+
+  /**
+   * Init JobMaster. To initiate or recover other components(like metrics and extra coordinators).
+   *
+   * @param isRecover if it is recover from state
+   * @return init result
+   */
+  public Boolean init(boolean isRecover) {
+    LOG.info("Start to init job master. Is recover: {}.", isRecover);
+
+    if (this.runtimeContext.getExecutionGraph() == null) {
+      LOG.error("Init job master failed. Job graphs is null.");
+      return false;
+    }
+
+    // recover from last checkpoint
+    if (isRecover) {
+      // TODO (lianxin)
+    }
```

Comment:
Complete this TODO or remove this if block.

Reply:
Remove 'recover' for now. Will add back in 'checkpoint' part.